crates/net/network/src/session/active.rs

```diff
@@ -242,7 +243,7 @@ impl ActiveSession {
 
     /// Returns the deadline timestamp at which the request times out
     fn request_deadline(&self) -> Instant {
-        Instant::now() + self.request_timeout
+        Instant::now() + Duration::from_millis(self.request_timeout.load(Ordering::SeqCst))
```

Comment:
let's always use `Relaxed` ordering, we only really want atomic operations and do not necessarily care about ordering constraints